### ISSUE TITLE
Use same or relative path for backups when files will be modified/deleted: --suffix without --backup-dir

### ISSUE THREAD

**Issue by imthenachoman** [ncw/rclone#2801]
I think it would be awesome if `rclone` could make backups in the same folder as the original file and/or a folder relative to the same folder.

For example:

- backup `remote:drive:/path/to/file.txt` to `remote:drive:/path/to/file.txt.20181124`
- backup `remote:drive:/path/to/file.txt` to `remote:drive:/path/to/backups/file.txt.20181124`

I was hoping to add this enhancement to the code myself but I cannot figure out the code. I know, at minimum, lines 147-148 in `rclone/fs/config/configflags/configflags.go` would need to be modified to allow `--suffix` without `--backup-dir`. I can tell some other files need to be updated like `sync.go` but can't make sense of `rclone`'s code structure to know everywhere that would need to be updated. If someone is willing to point me in the right direction I would add and test the enhancement myself then submit a pull request.

**Comment by ncw** [comment]
Great - glad  you are having a go at this :-)

Apart from the config changes you'll need to set up the internals in `fs/sync/sync.go`

Here is where rclone sets up the backupDir

https://github.com/ncw/rclone/blob/eb4125376482206505bc9521408a5b58ad074c2b/fs/sync/sync.go#L108-L128

I think you want to add something like this on the end at line 128

```go
} else if fs.Config.Suffix != "" {
		// --backup-dir is not set but --suffix is - use the destination as the backupDir
		s.backupDir = fdst
		s.suffix = fs.Config.Suffix
		if !operations.CanServerSideMove(s.backupDir) {
			return nil, fserrors.FatalError(errors.New("can't use --suffix on a remote which doesn't support server side move or copy"))
		}
}
```

And that may be enough!

This will need a new test.  You may be able to modify the existing one

https://github.com/ncw/rclone/blob/eb4125376482206505bc9521408a5b58ad074c2b/fs/sync/sync_test.go#L1082

Let me know if you need more help :-)

**Comment by ncw** [comment]
> I think it would be awesome if `rclone` could make backups in the same folder as the original file and/or a folder relative to the same folder.

All the above was for "same folder".

I think relative to the same folder is probably a different issue - let's get "same folder" out the way first.

**Comment by imthenachoman** [comment]
Awesome. Thanks. I will play with it this weekend. And I agree on the relative folder comment. Baby steps.

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.49

Thank you @yparitcher for doing the work.
